activerecord/lib/active_record/attribute_methods.rb

```diff
@@ -167,6 +167,24 @@ def attribute_names
         end
       end
 
+      # Returns an Array of Strings that can be treated as column names. This
+      # includes attributes and attribute aliases.
+      #
+      #   class Person < ActiveRecord::Base
+      #   end
+      #
+      #   Person.effective_column_names
+      #   # => ["id", "created_at", "updated_at", "name", "age"]
+      def effective_column_names
```

Comment:
I think this method name is doesn't really explain what it's doing, but I haven't been able to come up with a better name yet. `safe_column_names`? `attribute_column_names`?